sdk/storage/src/table/cloud_table.rs

```diff
@@ -288,27 +322,86 @@ where
         let (headers, body) =
             check_status_extract_headers_and_body(future_response, StatusCode::OK).await?;
 
-        log::trace!("body == {:?}", std::str::from_utf8(&body));
-        let entities = serde_json::from_slice::<EntityCollection<T>>(&body)?;
-        *continuation = Continuation::try_from(&headers)?;
-        Ok(Some(entities.value))
+        Ok((continuation_token.query_path.as_str(), &headers, &body).try_into()?)
+    }
+
+    pub fn stream_get_all<'a, T>(
+        &'a self,
+    ) -> impl Stream<Item = Result<QueryResult<T>, AzureError>> + 'a
+    where
+        T: Serialize + DeserializeOwned + 'a,
+    {
+        #[derive(Debug, Clone)]
+        enum States {
+            Init,
+            Continuation(ContinuationToken),
+        };
+
+        futures::stream::unfold(
+            Some(States::Init),
+            move |state: Option<States>| async move {
+                log::debug!("state == {:?}", state);
+                let response = match state {
+                    Some(States::Init) => self.begin_get_all().await,
+                    Some(States::Continuation(continuation_token)) => {
+                        self.continue_execution(continuation_token).await
+                    }
+                    None => return None,
+                };
+
+                let response = match response {
+                    Ok(response) => response,
+                    Err(err) => return Some((Err(err), None)),
+                };
+
+                let continuation_token = match &response.continuation_token {
+                    Some(ct) => Some(States::Continuation(ct.to_owned())),
+                    None => None,
+                };
+
+                Some((Ok(response), continuation_token))
+            },
+        )
     }
 
     pub fn stream_query<'a, T>(
         &'a self,
-        query: Option<&'a str>,
-    ) -> impl Stream<Item = Result<Vec<TableEntity<T>>, AzureError>> + 'a
+        query: &'a str,
+    ) -> impl Stream<Item = Result<QueryResult<T>, AzureError>> + 'a
     where
         T: Serialize + DeserializeOwned + 'a,
     {
-        futures::stream::unfold(Continuation::start(), move |mut cont| async move {
-            log::debug!("cont == {:?}", cont);
-            match self.execute_query::<T>(query, &mut cont).await {
-                Ok(Some(segment)) => Some((Ok(segment), cont)),
-                Ok(None) => None,
-                Err(err) => Some((Err(err), cont)),
-            }
-        })
+        #[derive(Debug, Clone)]
```

Comment:
Why not share this states enum across the two functions?

Reply:
copypasta 🤷‍♂️ ... good catch 😄